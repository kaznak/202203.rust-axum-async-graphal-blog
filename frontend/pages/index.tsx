import Layout from 'components/layout'
import Link from 'next/link'

export function Home() {
  return (
    <Layout title="rust-axum-async-graphql-blog">
      <Link href={`/posts`}>list</Link>
    </Layout>
  )
}

export default Home
