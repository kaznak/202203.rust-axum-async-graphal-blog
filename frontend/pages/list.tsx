import Head from 'next/head'
import Link from 'next/link'
import Image from 'next/image'

import { request, gql } from 'graphql-request'
import useSWR from 'swr'

import { API_ENDPOINT } from 'configs/app'

import vercelLogoSvg from 'public/vercel.svg'

const query = gql`
  query List {
    list {
      slug
      title
    }
  }
`

type PostList = {
  list: {
    slug: string
    title: string
  }[]
}

export function List() {
  const { data, error } = useSWR<PostList>(query, (query) =>
    request(API_ENDPOINT, query)
  )

  return (
    <div className="flex flex-col items-center justify-center min-h-screen py-2">
      <Head>
        <title>List</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className="w-full flex-1 px-20">
        <h1 className="text-6xl font-bold border-b-8">List</h1>

        {error ? (
          <div>failed to load</div>
        ) : !data ? (
          <div>loading...</div>
        ) : (
          data.list.map((post) => (
            <li key={post.slug}>
              <Link href={`/posts/${post.slug}`}>{post.title}</Link>
            </li>
          ))
        )}
      </main>

      <footer className="flex items-center justify-center w-full h-24 border-t">
        <a
          className="flex items-center justify-center"
          href="https://vercel.com"
          target="_blank"
          rel="noopener noreferrer"
        >
          Powered by{' '}
          <Image
            src={vercelLogoSvg}
            alt="Vercel Logo"
            className="h-4 ml-2"
            width={70.75}
            height={16}
          />
        </a>
      </footer>
    </div>
  )
}

export default List
