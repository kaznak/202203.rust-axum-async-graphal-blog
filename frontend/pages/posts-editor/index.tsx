import Layout from 'components/layout'
import { Editor } from 'components/editor'

export function Post() {
  return (
    <Layout title="create post">
      <Editor />
    </Layout>
  )
}

export default Post
