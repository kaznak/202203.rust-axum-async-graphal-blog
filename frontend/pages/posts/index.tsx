import Link from 'next/link'

import { useBlogPostList } from 'hooks/useBlog'

import Layout from 'components/layout'

export function List() {
  const { data, error } = useBlogPostList()

  if (error) {
    return <div>failed to load</div>
  }
  if (!data) {
    return <div>loading...</div>
  }
  return (
    <Layout title="List">
      {data.list.map((post) => (
        <li key={post.slug}>
          <Link href={`/posts/${post.slug}`}>{post.title}</Link>
        </li>
      ))}
    </Layout>
  )
}

export default List
