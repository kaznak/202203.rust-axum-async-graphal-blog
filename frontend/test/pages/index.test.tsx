import React from 'react'
import { render } from 'test/testUtils/testingLibraryWrapper'
import { Home } from 'pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
