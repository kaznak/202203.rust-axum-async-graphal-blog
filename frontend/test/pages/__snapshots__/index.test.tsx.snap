// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-center min-h-screen py-2"
  >
    <main
      class="w-full flex-1 px-20"
    >
      <span
        class="mx-4"
      >
        <a
          href="/posts"
        >
          goto list
        </a>
      </span>
      <span
        class="mx-4"
      >
        <a
          href="/posts-editor"
        >
          create
        </a>
      </span>
      <h1
        class="text-6xl font-bold border-b-8"
      >
        rust-axum-async-graphql-blog
      </h1>
      <section>
        <a
          href="/posts"
        >
          list
        </a>
      </section>
    </main>
    <footer
      class="flex items-center justify-center w-full h-24 border-t"
    >
      <a
        class="flex items-center justify-center"
        href="https://vercel.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt="Vercel Logo"
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzAuNzUiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block;"
            />
          </div>
          <img
            alt="Vercel Logo"
            class="h-4 ml-2"
            decoding="async"
            height="16"
            src="https://test-file-stub"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            width="70.75"
          />
        </div>
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
